fitparse/encoder.py

```diff
@@ -0,0 +1,440 @@
+import os
+import re
+import struct
+
+import six
+
+from fitparse import FitFileDataProcessor, profile, utils
+from .records import Crc, DataMessage, DefinitionMessage, MessageHeader, FieldData, FieldDefinition
+from .utils import fileish_open, FitParseError
+
+
+class FitFileEncoder(object):
```

Comment:
It doesn't look like this class allows for creating chained FIT files (basically multiple FIT files concatenated together, see [the commit that added support for parsing them](https://github.com/dtcooper/python-fitparse/commit/e39980ad114f23e529e8aff2d6aa5be039748550)). Not saying it has to, just letting you know that it's a possibility since it's a bit of an obscure part of the spec.

In fact, if you want to support chained FIT files, it should probably be taken care of by higher-level code that basically just does an `itertools.chain(*fileishs)` for the fileishs written by this class.

Reply:
Exactly, the code itself allows to append to a file, too. Just the file has to be opened by the user code. So, chaining is the task for the user code. Should not burden the core code.

Reply:
I get that it shouldn't be done as part of the normal encoding since it's at a higher level than messing around with the bytes of the file, but that doesn't mean it can't be included in the library.

If it's a common operation, pushing it into user code just means that everyone is going to have to write their own implementation. A better way would be to provide all the functionality the user would need, but let them decide when to use it. For example, one way would be to have a small method `chain` on the class that does something like (untested):
```python
def chain(self, *fits):
    if not self.completed:
        raise Exception("Can't chain more files onto an incomplete FIT file")

    for fit in fits:
        if isinstance(fit, FitFile):
            data = fit._file
        else:
            data = fileish_open(fit, 'rb')

        while True:
            block = data.read(BLOCK_SIZE)
            if not block:
                break
            self._write(block)
```

This way the library can check for issues (like appending data to an incomplete fit datastream) as well as handle compatibility with library objects (the `FitFile` shim), but it doesn't complicate the rest of the encoding process in any way.

Reply:
You can already chain with this code:
```
with FitFileEncoder(open(fitname, 'ab') as fit:
   ...
with FitFileEncoder(open(fitname, 'ab') as fit:
   ...
```
I have not seen any particular support for chaining in the FIT SDK, I think they support it the same way.